modules/face/src/facemarkLBF.cpp

```diff
@@ -0,0 +1,1294 @@
+                          License Agreement
+               For Open Source Computer Vision Library
+                       (3-clause BSD License)
+Copyright (C) 2013, OpenCV Foundation, all rights reserved.
+Third party copyrights are property of their respective owners.
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+  * Redistributions of source code must retain the above copyright notice,
+    this list of conditions and the following disclaimer.
+  * Redistributions in binary form must reproduce the above copyright notice,
+    this list of conditions and the following disclaimer in the documentation
+    and/or other materials provided with the distribution.
+  * Neither the names of the copyright holders nor the names of the contributors
+    may be used to endorse or promote products derived from this software
+    without specific prior written permission.
+This software is provided by the copyright holders and contributors "as is" and
+any express or implied warranties, including, but not limited to, the implied
+warranties of merchantability and fitness for a particular purpose are
+disclaimed. In no event shall copyright holders or contributors be liable for
+any direct, indirect, incidental, special, exemplary, or consequential damages
+(including, but not limited to, procurement of substitute goods or services;
+loss of use, data, or profits; or business interruption) however caused
+and on any theory of liability, whether in contract, strict liability,
+or tort (including negligence or otherwise) arising in any way out of
+the use of this software, even if advised of the possibility of such damage.
+
+This file was part of GSoC Project: Facemark API for OpenCV
+Final report: https://gist.github.com/kurnianggoro/74de9121e122ad0bd825176751d47ecc
+Student: Laksono Kurnianggoro
+Mentor: Delia Passalacqua
+*/
+
+#include "opencv2/face.hpp"
+#include "precomp.hpp"
+#include "liblinear.hpp"
+#include <fstream>
+#include <cmath>
+#include <ctime>
+#include <cstdio>
+#include <cassert>
+#include <cstdarg>
+
+namespace cv {
+namespace face {
+
+    #define TIMER_BEGIN { double __time__ = (double)getTickCount();
+    #define TIMER_NOW   ((getTickCount() - __time__) / getTickFrequency())
+    #define TIMER_END   }
+
+    #define SIMILARITY_TRANSFORM(x, y, scale, rotate) do {            \
+        double x_tmp = scale * (rotate(0, 0)*x + rotate(0, 1)*y); \
+        double y_tmp = scale * (rotate(1, 0)*x + rotate(1, 1)*y); \
+        x = x_tmp; y = y_tmp;                                     \
+    } while(0)
+
+    FacemarkLBF::Params::Params(){
+
+        cascade_face = "../data/haarcascade_frontalface_alt.xml";
+        shape_offset = 0.0;
+        n_landmarks = 68;
+        initShape_n = 10;
+        stages_n=5;
+        tree_n=6;
+        tree_depth=5;
+        bagging_overlap = 0.4;
+        model_filename = "LBF.model";
```

Comment:
this is just the default filename for the saving file, is it also problem? if it were for reading file like the cascade face then for sure it will be problem

Reply:
This is minor problem (at least there is no relative path to subdirectory).